### ISSUE TITLE
Why does my next sequence start from the previous last state?

### ISSUE THREAD

**Issue by ammar-n-abbas** [hmmlearn/hmmlearn#435]
## I am reporting a bug

I am using Gaussian HMM with hmmlearn library in python. I have multiple sequences and I am modeling a left-right HMM topology:

```
num_states = 15
remodel = hmm.GaussianHMM(n_components=num_states,
                          n_iter=500,
                          verbose=True,
                          init_params="cm", params="cmt")

transmat = np.zeros((num_states, num_states))
# Left-to-right: each state is connected to itself and its
# direct successor.
for i in range(num_states):
    if i == num_states - 1:
        transmat[i, i] = 1.0
    else:
        transmat[i, i] = transmat[i, i + 1] = 0.5

# Always start in first state
startprob = np.zeros(num_states)
startprob[0] = 1.0

remodel.startprob_ = startprob
remodel.transmat_ = transmat
remodel = remodel.fit(df_hmm, lengths)
```

The state distribution should start with the initial state of 0, however, it starts with the last state of the sequence, is there something that I am missing?

[![first sequence][1]][1]
[![second sequence][2]][2]
[![third sequence][3]][3]

Thank you

  [1]: https://i.stack.imgur.com/MXQBz.png
  [2]: https://i.stack.imgur.com/KHpsL.png
  [3]: https://i.stack.imgur.com/TQEt5.png

**Comment by anntzer** [comment]
Your example is incomplete: df_hmm is not provided.

**Comment by ammar-n-abbas** [comment]
It is a sequence of sensor observations:

just a snapshot of it
![image](https://user-images.githubusercontent.com/28593585/143280865-333fd478-5e22-4dfd-a20a-345eef2d9953.png)

It is an array of (20631,14), where 14 represents the sensor observations at every state, and 20631 the total length of multiple sequences

**Comment by anntzer** [comment]
Again, with no data, I cannot reproduce the issue.

**Comment by ammar-n-abbas** [comment]
Hi Antony,

Sorry for that, please find the code and the dataset at this repo.
https://github.com/ammar-n-abbas/HMM_hmmlearn_multiSeq_multiObs_leftRight_NASAdatset

Thank you for your help

**Comment by anntzer** [comment]
Again, this is not a minimal example.  I don't know what the gym, lib, stable_baselines modules are, I don't want to use tensorflow or keras, etc.

**Comment by ammar-n-abbas** [comment]
Sorry, I have updated the file

**Comment by ammar-n-abbas** [comment]
[train_FD001.txt](https://github.com/hmmlearn/hmmlearn/files/7603640/train_FD001.txt)


`############################################################################################################
#                                           IMPORTING LIBRARIES
# ############################################
[train_FD001.txt](https://github.com/hmmlearn/hmmlearn/files/7603640/train_FD001.txt)
##############################################################

import pandas as pd
import numpy as np
import os
import matplotlib.pyplot as plt

from sklearn.preprocessing import StandardScaler, MinMaxScaler
from hmmlearn import hmm


standard = StandardScaler()
minmax = MinMaxScaler()

############################################################################################################
#                                           DATASET PREPARATION
# ##########################################################################################################

dir_path = os.getcwd()
dataset = 'train_FD001.txt'
df = pd.read_csv(dir_path + r'/CMAPSSData/' + dataset, sep=" ", header=None, skipinitialspace=True).dropna(axis=1)
df = df.rename(columns={0: 'unit', 1: 'cycle', 2: 'W1', 3: 'W2', 4: 'W3'})
df_A = df[df.columns[[0, 1]]]
df_S = df[df.columns[list(range(5, 26))]]


############################################################################################################
# **********************************************************************************************************
#                                       HIDDEN MARKOV MODEL (LIBRARY)
# **********************************************************************************************************
# ##########################################################################################################


df_hmm = minmax.fit_transform(df_S)

df_hmm = pd.DataFrame(df_hmm)
cols_to_drop = df_hmm.nunique()[df_hmm.nunique() == 1].index
df_hmm = df_hmm.drop(cols_to_drop, axis=1)
cols_to_drop = df_hmm.nunique()[df_hmm.nunique() == 2].index
df_hmm = df_hmm.drop(cols_to_drop, axis=1).to_numpy()

lengths = [df[df['unit'] == i].cycle.max() for i in range(1, df_A['unit'].max() + 1)]

num_states = 15
remodel = hmm.GaussianHMM(n_components=num_states,
                          n_iter=500,
                          verbose=True,
                          init_params="cm", params="cmt")

transmat = np.zeros((num_states, num_states))

# Left-to-right: each state is connected to itself and its direct successor.
for i in range(num_states):
    if i == num_states - 1:
        transmat[i, i] = 1.0
    else:
        transmat[i, i] = transmat[i, i + 1] = 0.5

# Always start in first state
startprob = np.zeros(num_states)
startprob[0] = 1.0

remodel.startprob_ = startprob
remodel.transmat_ = transmat
remodel = remodel.fit(df_hmm, lengths)

state_seq = remodel.predict(df_hmm)
pred = [state_seq[df[df['unit'] == i].index[0]:df[df['unit'] == i].index[-1] + 1] for i in
        range(1, df_A['unit'].max() + 1)]


plt.figure(0)
plt.plot(pred[0])
plt.xlabel('# Flights')
plt.ylabel('HMM states')

plt.figure(1)
plt.plot(pred[1])
plt.xlabel('# Flights')
plt.ylabel('HMM states')

plt.figure(2)
plt.plot(pred[2])
plt.xlabel('# Flights')
plt.ylabel('HMM states')

plt.show()`
